### PR TITLE
Adding the plans tab in the project's dashboard

### PR DESCRIPTION
**Pull request by igorrfc**
This tab allows the project owner and/or an admin user associates plans to their projects.

implemented-plans-tab

### CODE REVIEW

File: app/assets/stylesheets/juntos_bootstrap/_main.scss [hunk 1]
```diff
@@ -500,6 +500,39 @@ p {
...
+    cursor: pointer;
+
+    .check-box {
+      position: absolute;
+      top: 50%;
+      transform: translateY(-50%);
+    }
+
+    .content {
+      height: 120px;
```
Comment (marclerodrigues) [inline app/assets/stylesheets/juntos_bootstrap/_main.scss]: You should use some scss linter and organize your properties alphabetically.

File: app/models/project.rb [hunk 2]
```diff
@@ -374,4 +377,8 @@ def has_minimum_goal_for_video?
   def reject_project_image(attributes)
     new_record? && attributes[:original_image_url].nil?
   end
+
+  def can_has_associated_plans?
```
Comment (marclerodrigues) [inline app/models/project.rb]: Avoid prefixing methods with "can_".

File: app/assets/stylesheets/juntos_bootstrap/_main.scss (hunk 1 above)
Comment (marclerodrigues) [inline app/assets/stylesheets/juntos_bootstrap/_main.scss]: Also, what about using em as unit?

File: app/models/project.rb (hunk 2 above)
Comment (igorrfc) [inline app/models/project.rb]: Can u give another example which fits in this case, @marclerodrigues ? Sounds a bit weird for me too, but I can't think in other name.

File: app/assets/stylesheets/juntos_bootstrap/_main.scss (hunk 1 above)
Comment (igorrfc) [inline app/assets/stylesheets/juntos_bootstrap/_main.scss]: Do you mean use `em` instead `px` for height?

Comment (marclerodrigues) [inline app/assets/stylesheets/juntos_bootstrap/_main.scss]: Yes, and not only for the height but for all properties that you're changing.

File: app/models/project.rb [hunk 3]
```diff
@@ -31,6 +31,7 @@ class Project < ActiveRecord::Base
   belongs_to :user
   belongs_to :category
   has_and_belongs_to_many :channels
+  has_and_belongs_to_many :plans, join_table: 'projects_plans'
```
Comment (marclerodrigues) [inline app/models/project.rb]: You have no tests in your model spec for these changes about has_and_belogns_to_many relation.
```ruby
it { is_expected.to have_and_belong_to_many(:plans)
```

File: app/models/plan.rb [hunk 4]
```diff
@@ -9,4 +9,6 @@ class Plan < ActiveRecord::Base
             multiple: true, i18n_scope: "enumerize.plan.payment_methods"
 
   validates_presence_of :plan_code, :name, :amount, :payment_methods
+
+  has_and_belongs_to_many :projects, join_table: 'projects_plans'
```
Comment (marclerodrigues) [inline app/models/plan.rb]: The same about the specs for this relation.

File: app/models/project.rb (hunk 2 above)
Comment (marclerodrigues) [inline app/models/project.rb]: What do you mean by "another example" :smile: ? You can check the ruby style guide.
https://github.com/bbatsov/ruby-style-guide/blob/master/README.md#naming

Comment (igorrfc) [inline app/models/project.rb]: I just asked you for a name suggestion, u know? 😄 
what u thought about `permit_association_with_plans?`  ?

File: config/locales/catarse_bootstrap/views/projects/show.en.yml [hunk 5]
```diff
@@ -118,11 +123,13 @@ en:
       contributions: 'donors'
       goal: achieved of %{total}
     show:
+      no_plans_available: 'Nenhum plano está disponível'
```
Comment (Maychell) [inline config/locales/catarse_bootstrap/views/projects/show.en.yml]: Isso não deveria estar em en?

Comment (igorrfc) [inline config/locales/catarse_bootstrap/views/projects/show.en.yml]: eita, pode crer kkk

File: config/locales/catarse_bootstrap/views/projects/show.en.yml [hunk 6]
```diff
@@ -29,6 +29,11 @@ en:
         excerpt: 'Do you have any company that supports your fundraising campaign? If you have, put them right here! You can put up to 3 logos that will be displayed to the right side of the project. Remember, supporters must be of your project, not your NGO as a whole.'
         hint: 'We recommend 264x75 pixels images.'
         link: "Partner's link"
+    dashboard_project_plans:
+      title: 'What donate plans are available for your project?'
```
Comment (Maychell) [inline config/locales/catarse_bootstrap/views/projects/show.en.yml]: `title: 'Which donate plans are available for your project?'`

File: spec/models/project_spec.rb [hunk 7]
```diff
@@ -41,6 +42,28 @@
...
+      context "when the project is not recurring" do
+        let(:channel) { create(:channel) }
+        let(:plan) { create(:plan) }
+        subject { build(:project, channels: [channel], plans: [plan]) }
+
+        it "cannot has associated plans" do
+          expect(subject).to be_invalid
+        end
+      end
+    end
```
Comment (Maychell) [inline spec/models/project_spec.rb]: Acho que o teste está repetindo algumas coisas que não fazem tanta parte do escopo da asserção.
Talvez se movesse essa parte repetida para cima e deixasse só o que realmente muda dentro do `context`, ficaria melhor de ler.

O que acha disso?
```Ruby
describe "plans" do
	let(:plan) 		{ create(:plan) }
	let(:project) { build(:project, channels: [channel], plans: [plan]) }

	context "when the project is in recurring channel" do
		let(:channel) { create(:channel, :recurring) }

		it "can have associated plans" do
			expect(project).to be_valid
		end
	end
		
	context "when the project is not in recurring channel" do
		let(:channel) { create(:channel) }

		it "cannot have associated plans" do
			expect(project).to be_invalid
		end
  end
end
```

File: app/models/project.rb (hunk 2 above)
Comment (marclerodrigues) [inline app/models/project.rb]: It's a good name. Let's keep it. :+1:

**Pull request merged**